src/pages/playground/Playground.tsx

```diff
@@ -346,9 +440,42 @@ const Playground: React.FC<PlaygroundProps> = props => {
   }, [editorSessionId]);
 
   const hash = isSicpEditor ? props.initialEditorValueHash : location.hash;
+  const { uuid } = useParams();
 
   useEffect(() => {
-    if (!hash) {
+    const handleURL = (uuid: string | undefined) => {
+      if (uuid !== undefined) {
+        fetch(`${Constants.backendUrl}/api/shared_programs/${uuid}`)
+          .then(response => response.json())
+          .then(resp => {
+            // console.log("resp", resp)
+            const res: Partial<ShareLinkState> = new ShareLinkStateDecoder(resp).decodeWith(
+              new JsonDecoderDelegate()
+            );
+            // console.log('decode', res);
+            resetConfig(
+              res,
+              { handleChangeExecTime, handleChapterSelect },
+              workspaceLocation,
+              dispatch,
+              fileSystem
+            );
+          });
+      } else {
+        const config = new ShareLinkStateDecoder(location.hash).decodeWith(
+          new UrlParamsDecoderDelegate()
+        );
+        resetConfig(
+          config,
+          { handleChangeExecTime, handleChapterSelect },
+          workspaceLocation,
+          dispatch,
+          fileSystem
+        );
+      }
+    };
```

Comment:
Let's move this function out of `useEffect` to prevent unnecessary recreation. Also please remove the commented code, thanks!

Reply:
move this out and wrapped it with `useCallback`